### DISCUSSION TITLE
Android record audio without too much background noice

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#4720] [General Discussions]
### Describe the project you are working on

2d and 3d

### Describe the problem or limitation you are having in your project

I can record audio from my user on Android. But my problem is while they play the record there are too much background noice which make the audio disaster. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

May it can improve by adding noise reduction system. Or giving a frequent limit like it won't record sounds under 2db . 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

No idea

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No idea

### Is there a reason why this should be core and not an add-on in the asset library?

It should be on core. Because audio stream player can handle this

**Comment by Calinou** [comment]
Is this for real-time purposes, or are you OK with having the audio denoised once the user stops recording the audio?

In general, I'd say audio denoising is better suited to an add-on, as it's quite complex if you want to do it in real-time.

**Comment by ghost** [comment]
> Is this for real-time purposes, or are you OK with having the audio denoised once the user stops recording the audio?
> 
> In general, I'd say audio denoising is better suited to an add-on, as it's quite complex if you want to do it in real-time.


I want to make game like free fire where user is saying something to collaborate with other players . And it is important to remove Extra background noise from the audio
